src/libraries/System.Private.CoreLib/src/System/IO/Directory.Unix.cs

```diff
@@ -20,5 +25,24 @@ private static DirectoryInfo CreateDirectoryCore(string path, UnixFileMode unixC
 
             return new DirectoryInfo(path, fullPath, isNormalized: true);
         }
+
+        private static unsafe string CreateTempSubdirectoryCore(string? prefix)
+        {
+            // mkdtemp takes a char* and overwrites the XXXXXX with six characters
+            // that'll result in a unique file name.
+            string template = $"{Path.GetTempPath()}{prefix}XXXXXX\0";
+            byte[] name = Encoding.UTF8.GetBytes(template);
+
+            // Create the temp directory.
+            byte* result = Interop.Sys.MkdTemp(name);
+            if (result == null)
```

Comment:
This pointer is potentially pointing into garbage at this point, since it's the same pointer that was passed in to mkdtemp but the array is no longer pinned and could have been moved.  You're not actually using the pointer other than to see if it's null, so it's safe, but I'd suggest not even storing it into a local in order to discourage any such use, e.g.
```suggestion
            if (Interop.Sys.MkdTemp(name) is null)
```

Reply:
Also, it would be nice for `MkdTemp` to take the unmanaged pointer as the argument and not depend on pinning via PInvoke marshalling for the same reason.

Reply:
Done. I also changed Path.GetTempFileName, since this code was cloned from there.